src/digitalIdentity-backend/src/main/java/didentity/amos/digitalIdentity/controller/AuthenticationController.java

```diff
@@ -71,44 +75,46 @@ public class AuthenticationController {
 
 
     @PostMapping(path="/update")
-    public @ResponseBody String update(@RequestParam String name, @RequestParam String surname, @RequestParam String birthday,
-    @RequestParam String email, @RequestParam String company, @RequestParam String team, @RequestParam String user_role) {
-        Iterable<User> users = userRepository.findAll();
+    public @ResponseBody ResponseEntity<String> update(@RequestParam Integer id, @RequestParam String name, @RequestParam String surname, @RequestParam(required=false) String birthday,
+    @RequestParam String email, @RequestParam(required=false) String company, @RequestParam(required=false) String team, @RequestParam(required=false) String user_role) {
+        LinkedList<Integer> ids = new LinkedList<Integer>();
+        ids.add(id);
+        Iterable<User> DIs = userRepository.findAllById(ids);
         
-        // TODO Jannik: findAll() ist ziemlich inperformant; Ich wusste leider nicht wie man e_mail und password direkt im Repository abfragen kann
-        for (User userDatabase : users) {
-            if(userDatabase.getEmail().equals(email)) {
-
-                User user = userDatabase;
-                user.setName(name);
-                user.setSurname(surname);
-                user.setBirthday(birthday);
-                user.setEmail(email);
-                user.setCompany(company);
-                user.setTeam(team);
-
-                switch (user_role) {
-                    case "admin":
-                        user.setUserRole(UserRole.fromString("ROLE_ADMIN"));
-                        break;
-                    case "employee":
-                        user.setUserRole(UserRole.fromString("ROLE_EMPLOYEE"));
-                        break;
-                    case "hr_employee":
-                        user.setUserRole(UserRole.fromString("ROLE_HR_EMPLOYEE"));
-                        break;
-                    case "guest":
-                        user.setUserRole(UserRole.fromString("ROLE_GUEST"));
-                        break;
-                    default:
-                        return "\"user role not recognized!\"";
-            }
-            
-            userRepository.save(user);
-            return "\"success\"";
+        Iterator<User> diIterator = DIs.iterator();
+        if (!diIterator.hasNext()) {
+            return ResponseEntity.status(500).body("\"No DI with this id was found!\"");
+        }
+        User firstDI = diIterator.next();
+
+        if (name != null && name != "")         { firstDI.setName(name); }
```

Comment:
Man kann hier mit Optional<T> arbeiten, dann muss man nicht immer in if-Anweisungen abfragen, ob die Variablen null oder leer sind. Funktioniert so wie im Moment aber auch